shared/api/v3/course.ts

```diff
@@ -0,0 +1,160 @@
+    courseWaitlistLength: null;
+    courseEnrollmentStatus: string;
+    permCount: integer;
+}
+
+export interface Schedule {
+    scheduleDays: string;
+    scheduleStartTime: string;
+    scheduleEndTime: string;
+    scheduleStartDate: string;
+    scheduleEndDate: string;
+    scheduleTermCount: integer;
+    scheduleTerms: integer[];
+    scheduleLocation: string;
+}
+
+export function courseFromV4Section(s: APIv4.Section): Course {
+    const sectionCodeString = APIv4.stringifySectionCode(s.identifier);
+    const courseCodeString = APIv4.stringifyCourseCode(s.course.code);
+
+    const startDate = stringifyCourseDate(s.startDate);
+    const endDate = stringifyCourseDate(s.endDate);
+
+    const { termCount, terms } = getSectionTermFractions(s.identifier);
+
+    return {
+        courseCode: sectionCodeString,
+        courseName: s.course.title,
+        courseSortKey: [sectionCodeString],
+        courseMutualExclusionKey: [courseCodeString],
+        courseDescription: s.course.description,
+        courseInstructors: s.instructors.map((inst) => inst.name),
+        courseTerm: stringifySectionTermCode(s.identifier),
+        courseSchedule: s.schedules.flatMap((schedule) => {
+            const days = stringifyWeekdayEnumArray(schedule.days);
+            const startTime = stringifyTime(schedule.startTime);
+            const endTime = stringifyTime(schedule.endTime);
+            return schedule.locations.map((location) => ({
+                scheduleDays: days,
+                scheduleStartTime: startTime,
+                scheduleEndTime: endTime,
+                scheduleStartDate: startDate,
+                scheduleEndDate: endDate,
+                scheduleTermCount: termCount,
+                scheduleTerms: terms,
+                scheduleLocation: location,
+            }));
+        }),
+        courseCredits: s.credits,
+        courseSeatsTotal: s.seatsTotal,
+        courseSeatsFilled: s.seatsFilled,
+        courseWaitlistLength: null,
+        courseEnrollmentStatus: stringifySectionStatusEnum(s.status),
+        permCount: s.permCount,
+    };
+}
+
+export function courseListFromV4SectionList(sections: APIv4.Section[]): {
+    terms: [];
+    courses: Course[];
+} {
+    return {
+        terms: [],
+        courses: sections.map(courseFromV4Section),
+    };
+}
+
+function stringifyWeekdayEnumArray(arr: APIv4.WeekdayEnum[]): string {
+    return arr.join("");
+}
+
+function stringifyTime(secs: integer): string {
+    if (secs < 0 || secs >= 24 * 60 * 60) {
+        throw Error(`Time out of range: ${secs}`);
+    }
+    const total_minutes = secs / 60;
+
+    if (!Number.isInteger(total_minutes)) {
+        throw Error(`Time too precise to fit in hours:minutes format: ${secs}`);
+    }
+
+    const hours = Math.floor(total_minutes / 60);
+    const minutes = total_minutes % 60;
+
+    const hours_string = hours.toString().padStart(2, "0");
+    const minutes_string = minutes.toString().padStart(2, "0");
```

Comment:
generally TS/JS code style prefers `camelCase` to `snake_case`; certainly we should do that here for consistency with all the other variable naming conventions in the existing codebase